src/Core/AccurateComparison.h

```diff
@@ -21,17 +21,19 @@ using namespace DB;
 template <typename A, typename B>
 bool lessOp(A a, B b)
 {
+    /// anything vs NaN
```

Comment:
is there any possibility that A and B are integer types ?

Reply:
Yes.